feat: add social sharing buttons for blog posts

Arijit-Char opened the pull request:
# feat(blog): add social sharing buttons for blog posts 

![image](https://github.com/user-attachments/assets/025776a1-9985-46ce-a99a-774482a5d5a0)

This PR implements **issue #3649** ([Feature Request #3649](https://github.com/asyncapi/website/issues/3649)) by adding social sharing buttons to blog posts, enabling users to easily share content on social media.

## Changes Implemented:
1. **Added Social Sharing Buttons** to each blog post for:  
   - **Twitter**  
   - **LinkedIn**  
   - **Facebook**  
   - Each button generates a pre-filled sharing link, including the blog post title, URL, and relevant hashtags.  

2. **Updated Files:**  
   - **`components/layout/BlogLayout.tsx`**  
     - Integrated `react-share` components for social sharing.  
     - Created dynamic share URLs using `post.slug` and `post.title`.  
   - **`package.json`**  
     - Added `react-share` package (`^5.2.2`).  

## How This Improves the Blog:
- Enhances content visibility by enabling easy sharing across social platforms.  
- Improves user engagement by providing an intuitive way to promote blog posts.  

## Breaking Changes:
- **Yes**, this introduces a new dependency (`react-share`). 



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - **Social Sharing:** Blog posts now feature sharing buttons, allowing users to effortlessly share content on popular social media platforms.
  - **Internationalization:** Expanded support lays the groundwork for enhanced language capabilities in future updates.
  - **Dependency Updates:** Added new dependencies for social sharing and internationalization support.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Arijit-Char committed:
Fix: [FEATURE] <Social Sharing Buttons on Blogs> #3649

asyncapi-bot committed:
Merge branch 'master' into Issue#3649

Arijit-Char commented:
Hi @derberg, @akshatnema, @anshgoyalevil, @Mayaleeeee, and @asyncapi-bot-eve,

I've opened PR #3946 to add social sharing buttons to blog posts. While the feature is working as expected, some checks are failing.

Could you please take a look and provide any insights on fixing them? Also, a review would be greatly appreciated! 🙌

Thanks in advance! 😊

akshatnema commented:
Hey @Arijit-Char ,
We had these sharing buttons before in the blog post, but it got into some trouble. We don't need these buttons on the Blog card, instead it should be present inside blog page.

Arijit-Char committed:
Updated Icon from Twitter to X and Add accessibility attributes to share buttons

Arijit-Char commented:
> Hey @Arijit-Char , We had these sharing buttons before in the blog post, but it got into some trouble. We don't need these buttons on the Blog card, instead it should be present inside blog page.

Hey @akshatnema ,

Got it! I’ll update the PR to move the sharing buttons inside the blog page instead of the blog card. I'm working on the changes now and will push the updates soon.

Thanks for the clarification!

Arijit-Char committed:
Sharing buttons inside the blog page

Arijit-Char commented:
Hey @akshatnema ,

I've updated the PR to move the sharing buttons inside the blog page as requested. The changes are now implemented and ready for review.

Let me know if any further improvements are needed. Thanks!

![image](https://github.com/user-attachments/assets/025776a1-9985-46ce-a99a-774482a5d5a0)

asyncapi-bot committed:
Merge branch 'master' into Issue#3649

Arijit-Char committed:
Refactored to use single-line props as required and removed the AddThis code.

Arijit-Char committed:
Merge branch 'Issue#3649' of https://github.com/Arijit-Char/website into Issue#3649

Arijit-Char commented on components/layout/BlogLayout.tsx:
```diff
@@ -80,6 +91,26 @@ export default function BlogLayout({
+                    hashtag='#AsyncAPI'>
+                    <FacebookIcon size={32} round />
+                  </FacebookShareButton>
+                </div>
```
I have refactored to use single-line props as required and removed the AddThis code.

Arijit-Char committed:
resolve npm CI issues by adjusting Node version and permissions

Arijit-Char committed:
Changes after npm ci

Arijit-Char committed:
cont..

Arijit-Char committed:
cont.

Arijit-Char committed:
Merge branch 'master' into Issue#3649

akshatnema commented on package.json:
```diff
@@ -98,7 +98,9 @@
     "react-dom": "^18",
     "react-ga": "^3.3.1",
     "react-gtm-module": "^2.0.11",
+    "react-i18next": "^15.4.1",
```
Why do we need this library?

akshatnema reviewed (commented):
Hey @Arijit-Char,
I liked the approach of having social buttons on the blog post, in your PR. My only concern with design is, it looks like we are giving social handles of the author of the blog, not of the blog post. Can your tweak the design little bit, to make it more pointing towards the blog?

<img width="720" alt="image" src="https://github.com/user-attachments/assets/179cb0b5-163f-4be6-9da7-1bf3060202ca" />

akshatnema commented:
@Arijit-Char Also, please resolve the conflicts inside your PR.

Mayaleeeee reviewed (commented):
@Arijit-Char thanks for working on this.

Why not move the social icons to the extreme end of the blog?

It's confusing the way it is right now because I'm not sure if the social icons are for the author or the blog itself.

So I would suggest you move it down, reduce its size, and let's see how it will turn out.

cc @akshatnema

akshatnema commented:
@Mayaleeeee I think we shouldn't move it to the end of the Blog post because it will then have very less appearance for the readers. Let's ensure that we pick the right place on the top of the blog post only, so as to have good number of shares of our blogs, in the social media.

anshgoyalevil commented:
Closing due to inactivity. Please reopen if you need to continue working on the PR
